tests/integration/azure_blob/test_azure_blob.py

```diff
@@ -32,90 +34,91 @@
 }
 
 
-def azure_connection_configured():
-    return config["env"].get("AZURE_STORAGE_CONNECTION_STRING") is not None
-
-
 def prepare_env(auth_method):
-    # Remove any previously existing env_vars needed for authentication
+    if not config["env"].get("AZURE_CONTAINER"):
+        return False
+
     for k, env_vars in AUTH_METHODS_AND_REQUIRED_PARAMS.items():
         for env_var in env_vars:
             os.environ.pop(env_var, None)
-    # Then set the env_var needed to test a specific authentication method
-    env_vars = AUTH_METHODS_AND_REQUIRED_PARAMS.get(auth_method)
-    for env_var in env_vars:
-        os.environ[env_var] = config["env"].get(env_var)
 
-
-@pytest.mark.skipif(
-    not azure_connection_configured(),
-    reason="This is an integration test, add the needed environment variables in test-azure-blob.yml "
-    "to run it",
+    test_params = AUTH_METHODS_AND_REQUIRED_PARAMS.get(auth_method)
+    if not test_params:
+        return False
+
+    for env_var in test_params:
+        env_value = config["env"].get(env_var)
+        if not env_value:
+            return False
+        os.environ[env_var] = env_value
+
+    logger.info(f"Testing auth method {auth_method}")
+    return True
+
+
+pytestmark = pytest.mark.parametrize(
+    "auth_method",
+    [
+        pytest.param(
+            key,
+            marks=pytest.mark.skipif(
+                not prepare_env(key), reason=f"Auth method {key} not configured."
```

Comment:
I think let's rename to `try_set_auth_env_vars` or something similar (try hinting it will fail silently, and `prepare_env` sounds too wide, it just handles the auth env vars)